packages/react-core/src/components/TimePicker/TimePickerUtils.tsx

```diff
@@ -90,3 +92,16 @@ export const validateTime = (time: string, timeRegex: RegExp, delimiter: string,
   // empty string is valid
   return time === '' || (timeRegex.test(time) && validHours);
 };
+
+export const getHours = (time: string, delimiter: string) => {
+  const parts = time.match(new RegExp(`(\\d+)${delimiter}(\\d+)(\\s?)(\\w*)`));
```

Comment:
Can you add capture groups to `TimePicker.tsx:getRegExp` and pass that to `getHours` and `getMinutes` instead? No need for yet another regex.